src/track/trackmetadatataglib.cpp

```diff
@@ -1676,110 +1713,208 @@ Result readTrackMetadataAndCoverArtFromFile(TrackMetadata* pTrackMetadata, QImag
                 }
             }
         }
-    } else {
-        qWarning() << "Unsupported file type" << fileType;
+        break;
+    }
+    default:
+        qWarning()
+            << "Cannot read metadata from file"
+            << fileName
+            << "with unknown or unsupported file type"
+            << fileType;
+        return ERR;
     }
 
     qWarning() << "Failed to read track metadata from file" << fileName;
     return ERR;
 }
 
-Result writeTrackMetadataIntoFile(const TrackMetadata& trackMetadata, QString fileName) {
-    const QString fileType(getFileTypeFromFileName(fileName));
+// Encapsulates subtle differences between TagLib::File::save()
+// and variants of this function in derived subclasses.
+class TagSaver final {
```

Comment:
Sorry, now it looks somehow half done, nothing that prevent merging, but it itches me to write: ;-) 

The original issue was: some code out of sight vs. code duplication. Now we have more code and still code out of sight. 
How about turning the TagSaver glass into a abstract base class and Implement it for each file type? 
This way we could meld don the whole case tree above. 
Alternative we can move the whole case tree to one initOnce() function or to a initOnce() function for each type.

This way we could get also around the unique pointer moving.